lib/vehicle_factory.rb

```diff
@@ -0,0 +1,34 @@
+class VehicleFactory
+    attr_accessor :vehicle_lot
+    
+    def initialize
+        @vehicle_lot = []
+    end
+
+    def create_vehicles(vehicle_data)
+        vehicle_data.each do |vehicle|
+            filtered_data = filter_data(vehicle)
+            @vehicle_lot << Vehicle.new(filtered_data)
+        end
+    end
+
+    def filter_data(data)
+        filtered_set = Hash.new("")
+        data.each_pair do |key, value|
+          if key.to_s.include?("vin") || key == :vin
+            filtered_set[:vin] = value
+          elsif key.to_s.include?("model_year") || key == :year
+            filtered_set[:year] = value
+          elsif key == :make
+            filtered_set[:make] = value
+          elsif key == :model || key.to_s.include?("body")
+            filtered_set[:model] = value
+          elsif key.to_s.include?("electric") || key == :engine
+            filtered_set[:engine] = :ev
+          elsif key.to_s.include?('fuel')
+            filtered_set[:engine] = value
+          end
+        end
+        filtered_set
+    end
```

Comment:
Nice job with all this logic. There are a lot of keys to check here. And to answer your question, there might be a way to further abstract all of this into one class that creates objects, but I'm not sure I would make that abstraction, could make everything more complex and harder to read.